content/posts/2019/2019-07-05-littlest-cuda-hub.md

```diff
@@ -0,0 +1,148 @@
+---
+title: "Setting up GPU Data Science environments for hackathons"
+subtitle: "AKA Littlest Cuda Hub"
+date: 2019-07-05T00:00:00+00:00
+draft: true
+author: "Jacob Tomlinson"
+categories:
+  - blog
+tags:
+  - rapids
+  - iris
+  - aws
+  - jupyterhub
+  - cuda
+  - gpu
+thumbnail: jupyter
+---
+
+## Background
+
+In my first week working at Nvidia I have been spending some time with my previous colleagues at the Met Office to explore how the two organisations can collaborate. 
+
+The theme for the week was to run a hackathon to explore how GPUs could be used to accelerate existing tools and workflows within the Met Office. To begin trying things out we needed a collaborative environment with access to GPUs. For this the Met Office kindly provided a GPU EC2 instance on AWS and we set up RAPIDS and the [Littlest JupyterHub](https://tljh.jupyter.org/en/latest/) (tljh) to expose it to collaborators.
+
+## Setup
+
+Let's walk through the steps to create our GPU enabled Littlest JupyterHub.
+
+### Infrastructure
+
+First we will need to build a ubuntu server with GPUs available. As the Met Office folks have provided access to an AWS account we will go through the excellent instructions from the [tljh docs](https://tljh.jupyter.org/en/latest/install/amazon.html) to set up on AWS with a few adjustments below.
```

Comment:
> First we will need to build a ubuntu server with GPUs available

It's not clear at this point why we need Ubuntu.  Follks may also not be familiar with TLJH, so we may want to spell things out here a bit.  

Maybe something like the following?:

1.  Our group was more familiar with Juptyer notebooks than with ssh'ing into an AWS machine, so we wanted to give them all Jupyter Notebooks
2.  To do this we wanted a JupyterHub setup and in particular wanted to try out TLJH, because we heard that it was particularly easy for this setting.  (they have great docs!)
3.  However because we wanted to use GPUs as well, we needed to deviate from TLJH docs a bit, this post now describes those necessary deviations....